apimodels/agent_models.go

```diff
@@ -93,7 +95,7 @@ type CreateHost struct {
 
 	// agent-controlled settings
 	CloudProvider       string `mapstructure:"provider" json:"provider" plugin:"expand"`
-	NumHosts            int    `mapstructure:"num_hosts" json:"num_hosts"`
+	NumHosts            string `mapstructure:"num_hosts" json:"num_hosts" plugin:"expand"`
```

Comment:
it feels super wrong to make/force callers (including from the rest v2 service to do this weird thing to our type system. 

Perhaps we should open a seperate ticket to make the expansion thing work with ints/

Reply:
Done https://jira.mongodb.org/browse/EVG-5571.